nest-city-account/src/bloomreach/bloomreach.service.ts

```diff
@@ -60,6 +61,76 @@ export class BloomreachService {
     return result
   }
 
+  async trackCustomer(
+    cognitoId: string,
+    {
+      email,
+      isIdentityVerified,
+      ...rest
+    }: {
+      email?: string
+      isIdentityVerified?: boolean
+      'custom:account_type'?: string
+    } & Omit<CognitoGetUserData, 'email' | 'sub' | 'idUser' | 'Enabled' | 'custom:account_type'>
+  ): Promise<boolean | undefined> {
+    const {
+      given_name: firstName,
+      family_name: lastName,
+      name,
+      UserCreateDate: registrationDate,
+      'custom:account_type': accountType,
+    } = rest
+
+    if (process.env.BLOOMREACH_INTEGRATION_STATE !== 'ACTIVE') {
+      return undefined
+    }
+    if (!cognitoId) {
```

Comment:
I don't think this is necessary, the type assures the `cognitoId` should not miss.